src/rep.rs

```diff
@@ -5,6 +5,48 @@ use chrono::{DateTime, Utc};
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(untagged)]
+pub enum ImageBuildChunk {
+    Stream {
+        stream: String,
+    },
+    Error {
+        error: String,
+        #[serde(rename = "errorDetail")]
+        error_detail: ErrorDetail,
+    },
+    Aux {
+        aux: Aux,
+    },
+}
```

Comment:
I think that `Update` instead of `Stream` and `Digest` instead of `Aux` would be more obvious.

Also, this is missing a `PullStatus` enum, for when you have to pull an image when you're building.

For example
```dockerfile
FROM alpine
ENV test=123
RUN echo 'Hello, World!'
```

```
~/src/personal/shiplift master*
❯ cargo run --example imagebuild -- test
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `target/debug/examples/imagebuild test`
Object({"stream": String("Step 1/3 : FROM alpine")})
Object({"stream": String("\n")})
Object({"id": String("latest"), "status": String("Pulling from library/alpine")})
Object({"id": String("4c0d98bf9879"), "progressDetail": Object({}), "status": String("Pulling fs layer")})
Object({"id": String("4c0d98bf9879"), "progress": String("[>                                                  ]  29.32kB/2.811MB"), "progressDetail": Object({"current": Number(29316), "total": Number(2811321)}), "status": String("Downloading")})
Object({"id": String("4c0d98bf9879"), "progress": String("[======>                                            ]    372kB/2.811MB"), "progressDetail": Object({"current": Number(371967), "total": Number(2811321)}), "status": String("Downloading")})
Object({"id": String("4c0d98bf9879"), "progress": String("[=========================================>         ]  2.325MB/2.811MB"), "progressDetail": Object({"current": Number(2324959), "total": Number(2811321)}), "status": String("Downloading")})
Object({"id": String("4c0d98bf9879"), "progressDetail": Object({}), "status": String("Download complete")})
Object({"id": String("4c0d98bf9879"), "progress": String("[>                                                  ]  32.77kB/2.811MB"), "progressDetail": Object({"current": Number(32768), "total": Number(2811321)}), "status": String("Extracting")})
Object({"id": String("4c0d98bf9879"), "progress": String("[============>                                      ]  688.1kB/2.811MB"), "progressDetail": Object({"current": Number(688128), "total": Number(2811321)}), "status": String("Extracting")})
Object({"id": String("4c0d98bf9879"), "progress": String("[==================================================>]  2.811MB/2.811MB"), "progressDetail": Object({"current": Number(2811321), "total": Number(2811321)}), "status": String("Extracting")})
Object({"id": String("4c0d98bf9879"), "progressDetail": Object({}), "status": String("Pull complete")})
Object({"status": String("Digest: sha256:08d6ca16c60fe7490c03d10dc339d9fd8ea67c6466dea8d558526b1330a85930")})
Object({"status": String("Status: Downloaded newer image for alpine:latest")})
Object({"stream": String(" ---> e50c909a8df2\n")})
Object({"stream": String("Step 2/3 : ENV test=123")})
Object({"stream": String("\n")})
Object({"stream": String(" ---> Running in 4eb710851786\n")})
Object({"stream": String("Removing intermediate container 4eb710851786\n")})
Object({"stream": String(" ---> dd2962e20dbc\n")})
Object({"stream": String("Step 3/3 : RUN echo \'Hello, World!\'")})
Object({"stream": String("\n")})
Object({"stream": String(" ---> Running in 6f32dc0a87d3\n")})
Object({"stream": String("Hello, World!\n")})
Object({"stream": String("Removing intermediate container 6f32dc0a87d3\n")})
Object({"stream": String(" ---> 6584f7c56a64\n")})
Object({"aux": Object({"ID": String("sha256:6584f7c56a6416a90f29708b84d25948a0f1d7498de8271bd73857121557e54d")})})
Object({"stream": String("Successfully built 6584f7c56a64\n")})
Object({"stream": String("Successfully tagged shiplift_test:latest\n")})
```

Reply:
> I think that `Update` instead of `Stream` and `Digest` instead of `Aux` would be more obvious.
I wanted to match original names but I guess renaming it to something more meaningful makes sense.

> Also, this is missing a `PullStatus` enum, for when you have to pull an image when you're building.
Yes, that's correct. And it actually turns out that `ImagePullChunk` is unnecessary because `ImageBuildChunk` can be used for both pulling and building if it has `PullStatus` as a variant.